src/datasets/arrow_dataset.py

```diff
@@ -2800,8 +2800,6 @@ def init_buffer_and_writer():
                     input_dataset = self.with_format(
                         self._format_type, columns=input_columns, output_all_columns=False, **self._format_kwargs
                     )
-                    if remove_columns:
-                        remove_columns = list(set(remove_columns) & set(input_columns))
```

Comment:
I guess this was the source of the issue with filter...

I'm wondering why this was introduced in:
- #2246

Reply:
I guess the contributor misinterpreted the purpose of `input_columns` and `remove_columns` in that PR.